### PR TITLE
[multi-cluster] Add a "Mesh" page

### PR DESCRIPTION
**Pull request by israel-hdez**
This adds a new top-level "Mesh" page which will contain data/information about the mesh structure when Istio is setup in a multi-cluster environment.

The scope of this PR, is to list the underlying kubernetes clusters where the mesh is built on top. 

image

Related kiali/kiali#3409

### CODE REVIEW

**Comment by israel-hdez** [comment]
@jshaughn We have been discussing this, so I think it makes sense that you are main the reviewer.
I'll also ask for a review from somebody from Jupiter.

File: src/routes.tsx
Comment (jshaughn) [inline src/routes.tsx:64]: Again, Lucas PR making some conflicting changes here.

File: src/types/Clustering.ts
Comment (jshaughn) [inline src/types/Clustering.ts:3]: What does this mean exactly?

**Review by jshaughn — COMMENTED** [review]
@israel-hdez This looks good for an initial list-based view of the mesh.  I added a couple of comments about why I think it should be set to DNM for now (just logistics).  Also, doesn't this PR require back-end changes?  I think you may need to add that labeling/info.

**Comment by israel-hdez** [comment]
> Also, doesn't this PR require back-end changes? I think you may need to add that labeling/info.

Yes, this is the back-end part: https://github.com/kiali/kiali-ui/pull/2000

File: src/types/Clustering.ts
Comment (israel-hdez) [reply src/types/Clustering.ts:3]: isHomeCluster is a flag to indicate that this is the cluster where Kiali/istio is installed.

**Comment by israel-hdez** [comment]
@jshaughn I did the renamings. Also, did the renamings in front-end part.
Once I get the approvals, I'll close this PR and open a new one targeting a new branch that we can use to work on the MC epic (and also I'll fix conflicts there).

**Comment by israel-hdez** [comment]
This is now targeting a new feature branch.
Conflicts are fixed.

**Review by jshaughn — APPROVED** [review]
LG

**Comment by israel-hdez** [comment]
Thanks @jshaughn . Merging.

**Pull request merged**
